rules/utils/is-function-self-used-inside.js

```diff
@@ -0,0 +1,43 @@
+'use strict';
+const {findVariable} = require('eslint-utils');
+
+const getReferences = (scope, nodeOrName) => {
+	const {references = []} = findVariable(scope, nodeOrName) || {};
+	return references;
+};
+
+/**
+Check if `this`, `arguments`, or `id` of function is used inside it self.
+
+@param {Node} node - The function node.
+@param {Scope} [scope] - The scope of the function node.
```

Comment:
This is marked as optional, but doesn't look like it is.

Reply:
Right, I'll fix.

Reply:
Another thing in this function, I ask to pass two arguments, but acutally `functionNode` is alway `functionScope.block`, I just don't want people get the wrong scope. 

The scope can also get by calling `scopeManager.acquire(node)`, but in rule if it's in function, `context.getScope()` is easier to get the function scope, so I was considering accept `scope/scopeManager` as the second parameter.

Any better idea?

Reply:
I think what you already have is the best solution.